iothub/device/src/Transport/HttpTransportHandler.cs

```diff
@@ -427,6 +421,29 @@ private static TimeSpan GetInvokeDeviceMethodOperationTimeout(MethodInvokeReques
             return customHeaders;
         }
 
+        private static object ConvertToMessageSystemProperty(string messagePropertyName, IEnumerable<string> headerValue)
+        {
+            string propertyValue = headerValue?.First();
+
+            if (messagePropertyName.Equals(MessageSystemPropertyNames.EnqueuedTime, StringComparison.Ordinal)
+                || messagePropertyName.Equals(MessageSystemPropertyNames.ExpiryTimeUtc, StringComparison.Ordinal))
+            {
+                return DateTime.TryParse(propertyValue, out DateTime dateTime) ? dateTime : (object)null;
+            }
+
+            if (messagePropertyName.Equals(MessageSystemPropertyNames.SequenceNumber, StringComparison.Ordinal))
+            {
+                return propertyValue == null ? 0 : Convert.ToUInt64(propertyValue, CultureInfo.InvariantCulture);
```

Comment:
This convert could throw if the format is incorrect. Either we should have a try catch and throw appropriately or if it is a scenario to ignore, we should try catch and continue with the method.

Reply:
Sequence no is not user-settable, so if service sets it to an invalid value, then the SDK will throw the exception. This is the current behavior.